MatrixSDK/Crypto/MXCrypto.m

```diff
@@ -1964,27 +1976,118 @@ - (MXHTTPOperation*)ensureOlmSessionsForDevices:(NSDictionary<NSString* /* userI
 
     if (devicesWithoutSession.count == 0)
     {
+        NSLog(@"[MXCrypto] ensureOlmSessionsForDevices: Have already sessions for all");
         if (success)
         {
             success(results);
         }
         return nil;
     }
 
+    
     NSString *oneTimeKeyAlgorithm = kMXKeySignedCurve25519Type;
-
-    // Prepare the request for claiming one-time keys
+    
+    // Devices for which we will make a /claim request
     MXUsersDevicesMap<NSString*> *usersDevicesToClaim = [[MXUsersDevicesMap<NSString*> alloc] init];
+    // The same but devices are listed by their identity key
+    NSMutableArray<NSString*> *devicesToClaim = [NSMutableArray array];
+    
+    // Devices (by their identity key) that are waiting for a response to /claim request
+    // That can be devices for which we are going to make a /claim request OR devices that
+    // already have a pending requests.
+    // Once we have emptied this array, we can call the success or the failure block. The
+    // operation is complete.
+    NSMutableArray<NSString*> *devicesInProgress = [NSMutableArray array];
```

Comment:
I had some difficulties to understand the meaning of this array. I think we should rename it by removing the suffix "InProgress"
I suggest: currentDevicesPool, fullDevicesList, currentDevices...

I would like to highlight this is the full list of the devices for this request whether or not we decide to make a /claim request here or waiting for an existing one

Reply:
well. This is a local variable. Its scope is thus well defined. The IDE colorises it accordingly.
I cannot find a better name for the moment.
 I am sticked on the fact that there are devices we are processing for this call. There will be a progression of the processing.Their number will decrease.

Reply:
I understood its usage - I'm still thinking a rename would be useful:
wdyt of "pendingDevicesList"?
If you disagree that's ok, this was just a suggestion